vault/request_handling.go

```diff
@@ -1687,10 +1687,12 @@ func (c *Core) handleLoginRequest(ctx context.Context, req *logical.Request) (re
 		// Attach the display name, might be used by audit backends
 		req.DisplayName = auth.DisplayName
 
-		// If this is not a role-based quota, we still need to associate the
-		// login role with this lease for later lease-count quotas to be
-		// accurate.
-		if reqRole == nil && resp.Auth.TokenType != logical.TokenTypeBatch {
+		loginRequest := resp.Auth.TokenType != logical.TokenTypeBatch
```

Comment:
I think this variable name might be misleading. We'll still be doing a login, only batch tokens don't have leases registered in the expiration manager, so we skip the `leaseEntry` creation and all that: https://github.com/hashicorp/vault-enterprise/blob/main/vault/request_handling.go#L2163

Maybe something like `requiresLease` is better?

Reply:
thanks!! good call